Create one avalon sdk for all clients(worker service, connector and requesters)

Ram-srini opened the pull request:
…ester)

1. Combine connector and client sdk to avalon sdk
2. Update all the application with the new sdk
3. Organize the utility modules inside the SDK.

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

manojgop commented on sdk/avalon_sdk/direct/jrpc/jrpc_util.py:
```diff
@@ -48,11 +26,12 @@ class JsonRpcErrorCode(Enum):
     UNSUPPORTED_MODE = 6
     KEY_EXISTS = 9
 
+def construct_message(status, message):
```
This method name looks very generic for the functionality it implements. May be rename it to create_json_status_message ?

Ram-srini replied:
Done

manojgop commented on sdk/avalon_sdk/ethereum/ethereum_worker_registry.py:
```diff
@@ -40,6 +41,112 @@ def __init__(self, config):
+            contract_file_name, contract_address
+        )
+
     def worker_register(
```
Can we organize the functions in file such that all public functions are followed by private functions or vice-versa. But let's follow a consistent ordering.

Ram-srini replied:
Does it really required?

manojgop commented on sdk/avalon_sdk/ethereum/ethereum_worker_registry.py:
```diff
@@ -0,0 +1,276 @@
+                details).buildTransaction(
+                    self.__eth_client.get_transaction_params()
+                )
+            tx = self.__eth_client.execute_transaction(txn_hash)
```
execute_transaction() function returns json status response. So better to name variable "tx" something like  "tx_status" to make the variable name look more meaningful

Ram-srini replied:
Done

manojgop commented on sdk/avalon_sdk/ethereum/ethereum_worker_registry.py:
```diff
@@ -0,0 +1,276 @@
+    def worker_register(
+            self, worker_id, worker_type, org_id, application_ids, details):
+        """
+        Register new worker with details of worker
```
Also mention the return value of the function. This function return status response formatted as JSON string.

Ram-srini replied:
Changed to return None or error and on success return txn receipt.

manojgop commented on sdk/avalon_sdk/interfaces/work_order_proxy.py:
```diff
@@ -13,42 +13,44 @@
+        This API is for proxy model.
+        params
+            ork_order_id is unique id to identify the work order request
+            ork_order_response is the Work Order response data in string
```
typo. ork_order_id -> work_order_id
ork_order_response -> work_order_response

Ram-srini replied:
Done

Ram-srini commented:
> lint_module sdk "$SINCE" 1 0 0 0 || rv=1

@danintel I have added  lint check for my new module and fixed the lint errors. Please review and approve.

Ram-srini commented:
@danintel Can you please approve this PR? This SDK PR required for proxy model implementation.

danintel commented:
Approved.

From: Ram-srini <notifications@github.com>
Sent: Thursday, February 6, 2020 9:31 AM
To: hyperledger/avalon <avalon@noreply.github.com>
Cc: Anderson, Daniel <daniel.anderson@intel.com>; Mention <mention@noreply.github.com>
Subject: Re: [hyperledger/avalon] Create one avalon sdk for all clients(worker service, connector and requesters) (#234)


@danintel<https://github.com/danintel> Can you please approve this PR? This SDK PR required for proxy model implementation.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/hyperledger/avalon/pull/234?email_source=notifications&email_token=AH26URJU7XFEU6ESFV2I4NLRBRCLZA5CNFSM4KO72MVKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELACPVY#issuecomment-583018455>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AH26UROQFBSP55FHEWTJJTLRBRCLZANCNFSM4KO72MVA>.
